src/plugins/intel_cpu/src/utils/shape_inference/shape_inference_ngraph.hpp

```diff
@@ -13,12 +13,14 @@ namespace intel_cpu {
 
 /**
  * This class wraps ngraph specific shape inference class to implement CPU plugin specific interface.
- * 
+ *
  */
 class NgraphShapeInfer : public IShapeInfer {
 public:
-    NgraphShapeInfer(std::shared_ptr<IShapeInferCommon> shape_infer, IShapeInfer::port_mask_t port_mask) :
-        m_shape_infer(shape_infer), m_port_mask(port_mask) {}
+    NgraphShapeInfer(std::shared_ptr<IStaticShapeInfer> shape_infer, IShapeInfer::port_mask_t port_mask) :
+        m_shape_infer(shape_infer), m_port_mask(port_mask) {
+            m_shape_infer->set_port_mask(port_mask);
+        }
```

Comment:
```suggestion
    NgraphShapeInfer(std::shared_ptr<IShapeInferCommon> shape_infer, IShapeInfer::port_mask_t port_mask) :
        m_shape_infer(shape_infer), m_port_mask(port_mask), m_is_new_infer_api{std::dynamic_pointer_cast<IStaticShapeInfer>(shape_infer) != nullptr}{}
```
Then in `NgraphShapeInfer ` infer function depends on m_is_new_infer_api flag use correct infer (or don't change ctor and use dynamic cast inside shape inference until old inference will be removed ticket #113186).

Then `infer` function could look like:
{
// check & prepare common data
if (m_is_new_infer_api){
  prepare data accessor
  do static cast
  call infer & collect results.
}else{
 prepare data map
call infer & collect results
}
...
}

The `NgraphShapeInfer` works like adapter between IShapeInfer and IShapeInferCommon/IStaticShapeInfer.
There is big chance that all these changes to `IShapeInferCommon/IStaticShapeInfer` and classes which implements them are not required.

Reply:
already revert modification of NgraphShapInfer and use dynamic cast inside shape inference